spikeextractors/extractors/nwbextractors/nwbextractors.py

```diff
@@ -364,16 +364,21 @@ def add_devices(recording: se.RecordingExtractor, nwbfile=None, metadata: dict =
         """
         if nwbfile is not None:
             assert isinstance(nwbfile, NWBFile), "'nwbfile' should be of type pynwb.NWBFile"
+        
         defaults = dict(
             name="Device",
-            description="Ecephys probe."
+            description="no description"
         )
-        if metadata is None or 'Device' not in metadata['Ecephys']:
-            metadata = dict(
-                Ecephys=dict(
-                    Device=[defaults]
-                )
-            )
+
+        if metadata is None:
+            metadata = dict() 
+
+        if 'Ecephys' not in metadata:
+            metadata['Ecephys'] = dict()
+
+        if 'Device' not in metadata['Ecephys']:
+            metadata['Ecephys']['Device'] = [defaults]
+
```

Comment:
While I don't like having to break a single if statement into 3 parts, I can see the issue here (though it's unmentioned in the PR post). If a user passes metadata that does not have a device field, this completely overwrites the entire dictionary as the metadata is working as a pointer at that level.

What we would really want to do here is a deep dictionary update - given that we'll soon be moving a lot of this over to nwb-conversion-tools, I'll OK this temporary change for now and we can refactor more cleanly once we gain access to all the nwb-conversion-tools utils.